packages/gatsby/content/advanced/lifecycle-scripts.md

```diff
@@ -15,7 +15,7 @@ Packages can define in the `scripts` field of their manifest various actions tha
 
 - **prepublish** is called before `yarn npm publish` and similar commands (even before the package has been packed). This is the place where you'll want to check that the project is in an ok state. Because it's only called on prepublish, **the prepublish hook shouldn't have side effects.** In particular don't transpile the package sources in prepublish, as people consuming directly your repository (such as through the [`git:` protocol](/features/protocols#git)) wouldn't be able to use your project.
 
-- **postinstall** is called after a package got successfully installed on the disk. It is guaranteed to be called in topological order (in other words, your dependencies' postinstalls will always run before yours). For backwards compatibility, **preinstall** and **install** are called as part of **postinstall**.
+- **postinstall** is called whenever the dependency tree of your workspace changes -- e.g. when any package is added, removed or changed. It is guaranteed to be called in topological order (in other words, your dependencies' postinstalls will always run before yours). For backwards compatibility, **preinstall** and **install** are called as part of **postinstall**.
```

Comment:
It's not limited to workspaces, if the dependency tree of a `Package` changes its `postinstall` will be run anew

Reply:
@merceyz: How about:
> is called whenever a dependency tree within the project changes -- e.g. when any package is added, removed or changed.

?

Reply:
Or just:
>is called whenever a dependency changes -- e.g. when any package is added, removed or changed.

Reply:
(Skimming https://yarnpkg.com/advanced/architecture for helpful language, but nothing is jumping out yet)

Reply:
(Skimming https://yarnpkg.com/advanced/architecture for helpful language, but nothing is jumping out yet)

Reply:
> is called after dependency tree changes are written to the disk -- e.g. after a package is added, removed or changed.

Based on https://yarnpkg.com/cli/install#details & https://yarnpkg.com/features/workspaces